SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -646,9 +646,13 @@ int64 USpatialActorChannel::ReplicateActor()
 
 			bCreatedEntity = true;
 
-			// If we're not offloading AND either load balancing isn't enabled or it is and we're spawning an Actor that we know
-			// will be load-balanced to another worker then preemptively set the role to SimulatedProxy.
-			if (!USpatialStatics::IsSpatialOffloadingEnabled() && !(SpatialGDKSettings->bEnableUnrealLoadBalancer && NetDriver->LoadBalanceStrategy->ShouldHaveAuthority(*Actor)))
+			// We preemptively set the Actor role to SimulatedProxy if:
+			//  - offloading is disabled (with offloading we never give up authority since we're always spawning authoritatively),
+			//  - load balancing is disabled (since the legacy behaviour is to wait until Spatial tells us we have authority) OR
+			//  - load balancing is enabled AND our lb strategy says this worker should have authority AND the Actor isn't locked.
+			if (!USpatialStatics::IsSpatialOffloadingEnabled() &&
+				(!SpatialGDKSettings->bEnableUnrealLoadBalancer
+					|| (!NetDriver->LoadBalanceStrategy->ShouldHaveAuthority(*Actor)) && !NetDriver->LockingPolicy->IsLocked(Actor)))
```

Comment:
I think this is something that should be discussed offline however it's worth calling out that this is giving the locking policy a very different role and one that I don't think entirely makes sense.
This may be wanted in the short term so I don't want to get in the way, but this highlights the need for us to work out what our lifecycles should be.

Reply:
I agree making this work risks users using the locking policy for things they should fix in the load-balancing strategy. But I think that's more a problem to fix in terms of guiding users on how to use the tools, and we should make locking work correctly here regardless